core/src/net/buffers/mod.rs

```diff
@@ -91,6 +92,14 @@ impl BufferConfig {
         buffer_config
     }
 
+    pub fn from_config_file<P>(path: P) -> BufferConfig
+        where P:Into<PathBuf>
+    {
+        let p: PathBuf = path.into();
+        let doc = HoconLoader::new().load_file(p).expect("Failed to load file").hocon().expect("Failed to load as HOCON");
```

Comment:
Please run `cargo fmt` before committing any code.

Reply:
Is it ok if I change `newline_style = "Auto"` in `rustfmt.toml`? Otherwise I will touch every file when running `cargo fmt` on Windows.